### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by BehindTheMath**
- Code cleanup
- Cleanup parse-options tests
- Clean up README
- Remove old switchFallback code

### CODE REVIEW

**Comment by BehindTheMath** [comment]
I'll take a look at those later.

BTW, you should also be able to push commits to this PR. Just checkout upstream/cleanup/misc-cleanup, and then push your commits back to the upstream branch.

**Comment by robinnorth** [comment]
Sure, thanks! I made comments rather than commits to avoid stepping on your toes, but if you like, I can action the changes I suggested. Equally happy to leave them to you, if you like.

**Comment by BehindTheMath** [comment]
Everything looks good, but I think we should leave this open until we're ready to release 0.2.5, in case there's anything we want to add.

**Comment by robinnorth** [comment]
Sounds good to me, hopefully we can get 0.2.5 out fairly soon!

**Comment by robinnorth** [comment]
I've actioned the items I suggested previously and rebased against master to keep this PR up-to-date. I can't think of anything else that needs cleaning up at present, other than possibly the description in `package.json` (and adding a `contributors` attribute?).

File: lib/eval-script.js [hunk 1]
```diff
@@ -13,12 +15,12 @@ module.exports = function(el) {
 
   script.type = "text/javascript"
 
-  if (src != "") {
-    script.src = src;
-    script.async = false; // force synchronous loading of peripheral js
+  if (src !== "") {
+    script.src = src
+    script.async = false // force asynchronous loading of peripheral js
```
Comment (BehindTheMath) [inline lib/eval-script.js]: @robinnorth I'm confused. Why did you change this back? I thought we agreed it was synchronous.

**Comment by MoOx** [comment]
(I think it's probably time for 1.0 ;) )

File: lib/eval-script.js (hunk 1 above)
Comment (robinnorth) [reply lib/eval-script.js]: Oops, I guess I didn't quite successfully resolve that merge conflict... Will fix.

**Comment by BehindTheMath** [comment]
@robinnorth What do you think about that last commit (2ec4c14)?

**Comment by BehindTheMath** [comment]
It looks like after this PR, all issues milestoned for 0.2.5 will be closed.

Is there anything else that needs to be done before working on releasing 0.2.5?

**Comment by robinnorth** [comment]
> @robinnorth What do you think about that last commit (2ec4c14)?

The library was obviously modularised with testability in mind, but these modules have never had corresponding tests, so it's probably OK to inline them like this.

> Is there anything else that needs to be done before working on releasing 0.2.5?

Only a version bump and any changes to `package.json` (I added you and I as contributors, hope that's OK) and then we're good to go! Do you have npm publish access?

**Comment by BehindTheMath** [comment]
>I added you and I as contributors, hope that's OK

@MoOx Should we add everyone who contributed (https://github.com/MoOx/pjax/graphs/contributors)?

>Do you have npm publish access

I do not. I'll update the changelog and version number, and then @MoOx can publish it.

**Comment by BehindTheMath** [comment]
>The library was obviously modularised with testability in mind

With that in mind, should we move out `loadUrl()`, `loadContent()`, and `afterAllSwitches()` to separate files?

**Comment by robinnorth** [comment]
> With that in mind, should we move out loadUrl(), loadContent(), and afterAllSwitches() to separate files?

It would be great to modularise and be able to independently test these methods, as they're basically the core of the whole library, but it's probably going to be a bit of an undertaking to do so right now. I think we should get this next release out and then add this as an aim for the next major milestone (and close #21 in the process).

**Comment by BehindTheMath** [comment]
Sounds good.

So we're just waiting on @MoOx for an answer about other contributors, and then we can merge this. I have a branch with the updated changelog and the version bump that I'll rebase and push once this is merged.

**Comment by MoOx** [comment]
Do whatever you feel appropriate.
I can give you npm access if you want or I can release it too. Just tell me what you prefer.

To make changelog easier I would recommend you in the future to only use "Squash and merge" option when merging PR. That helps to have a clean git history with kind of "one pr = one feature or a bugfix" history, which is easy to review and create release notes from.

**Comment by BehindTheMath** [comment]
>I can give you npm access if you want or I can release it too. Just tell me what you prefer.

If you can give me access, we won't have to bother you in the future. I'll DM you on Twitter.

>To make changelog easier I would recommend you in the future to only use "Squash and merge" option when merging PR. That helps to have a clean git history with kind of "one pr = one feature or a bugfix" history, which is easy to review and create release notes from.

I wasn't sure which to use, since it looks like in the past sometimes merge commits were used instead of squashing. Also, using merge commits doesn't require as much rebasing. But we can start squashing.

**Comment by MoOx** [comment]
You have been added to npm ;)

For the merging, it's a recommendation. At the time I made this lib, not sure it was a thing :D

**Comment by BehindTheMath** [comment]
>You have been added to npm ;)

Great, thanks.

>Do whatever you feel appropriate.

@robinnorth What do you think? Should we add anyone else?

**Comment by robinnorth** [comment]
@BehindTheMath it's up to you -- we could add/ask to add anyone who has ever made a contribution via a PR and add a note encouraging future contributors to add themselves to the list when making a PR, or we could just list contributors who have made a significant contribution (i.e. multiple PRs)/have commit access to the repo, or leave it as is. My personal inclination is that one PR doesn't really qualify for 'contributor' status, but you were here before me, so I'll defer to you on this 😄

**Comment by BehindTheMath** [comment]
I don't want to slight anyone by leaving them out, but at the same time I think it will get unwieldy if there's a huge list of everyone who submitted commit.

I'm going to leave it as it is, and if anyone wants, they can add themselves in.

**Comment by robinnorth** [comment]
I think that's a sensible approach 👍
